calling-apis/intro-to-asynchronous-programming.md

```diff
@@ -0,0 +1,136 @@
+
+In pain-staking detail, let's outline all the actions Mica would do, beginning with picking up the phone, and ending with eating the meal.
+
+<br/>
+
+<details style="max-width: 700px; margin: auto;">
+  <summary>
+    No, seriously. Take one minute to make a list of the actions Mica would do. After that, then read more here.
+  </summary>
+
+We may imagine that what Mica does is make the order, then wait for the order, and then eat.
+
+It's actually way more likely that Mica will make the order, then **before the order arrives, they do other things,** like read, shower, sleep, watch TV. Mica is able to eat their food only after the food order has been received, cooked, assembled, assigned a driver, driven over, and delivered.
+
+The food order was received, made, and delivered. However, while that was happening, Mica **did not wait for the food order to finish** in order to continue other actions. While waiting for the food, Mica didn't stop everything they were doing, sit on the couch, and stare at the wall. (Or maybe they did!)
+
+<br/>
+
+The point is, Mica didn't _need_ to stop executing other actions while waiting for the food. They continued their other actions and processes while the food-delivery process was working and finishing.
+
+Lastly, **once the food delivery _did_ happen**, Mica knew what action they had to do next: **receive the food** and then eat it!
+
+Throughout this lesson, we can keep in mind these two questions:
+
+1. When does this line of code "finish"?
+1. What do we do when this line of code "finishes"?
+
+</details>
+
+## Vocabulary and Synonyms
+
+| Vocab                    | Definition                                                                                                                                        | Synonyms          | How to Use in a Sentence                                                                                                                            |
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Flow of control          | The order of execution of programming statements.                                                                                                 | -                 | "When we trace code as it runs, we are following the flow of control"                                                                               |
+| Asynchronous programming | A method of programming that intentionally executes actions and processes outside of program flow, so that the program doesn't stop for a result. | Async programming | "Making API calls and loading pages on the web is usually asynchronous, and other code will execute without waiting for the result of an API call." |
```

Comment:
even with async code, there is a program flow. It just isn't necessarily the same order as the written lines of code.
stop alone sounds like the program could be terminating. add wait to stress that it is... waiting
```suggestion
| Asynchronous programming | A method of programming that intentionally executes actions and processes outside of the typical program flow, so that the program doesn't stop and wait for a result. | Async programming | "Making API calls and loading pages on the web is usually asynchronous, and other code will execute without waiting for the result of an API call." |
```